### DISCUSSION TITLE
Flux equation

### DISCUSSION THREAD

**Discussion by Irene-Moreno** [usnistgov/fipy#910] [General]
Hi,

I have some questions on how the diffusive flux from face to face is calculated in the software. Would you mind providing the equation that is used? Does it fulfill the Gibbs-Duhem relation, or alternatively, how is the mass balance ensured and the mass fraction in each term constrained to [0, 1]?

Thank you for your time!

Kind regards,
Irene

**Comment by guyer** [comment]
FiPy doesn't know anything about Gibbs-Duhem, per se, although you are free to pose your mass-balance equation in terms of chemical potential. See, e.g., [`examples.cahnHilliard.mesh2DCoupled`](https://www.ctcms.nist.gov/fipy/examples/cahnHilliard/generated/examples.cahnHilliard.mesh2DCoupled.html).

FiPy is a cell-centered finite volume code. As such, it solves a discretized version of the divergence equation, which means that fluxes are conserved by definition. See the [discretization of the `DiffusionTerm`](https://www.ctcms.nist.gov/fipy/documentation/numerical/discret.html#diffusion-term-nabla-cdot-left-gamma-1-nabla-phi-right).
